intTests/support/test-and-diff.sh

```diff
@@ -0,0 +1,223 @@
+# tests changes.
+#
+# It is fine to use this script when you have only one SAW script to
+# test. The infrastructure for supporting multiple scripts does not
+# cost much.
+#
+# There is, however, an expectation that if there are multiple scripts
+# in one test directory, they are all fast. Having multiple scripts at
+# once loses the ability to run them separately or address them
+# directly from enclosing test infrastructure. We don't want that to
+# incur significant costs or create aggravations.
+#
+# Note: we could, like make, not rerun tests whose output is newer
+# than (all) their inputs. So far I haven't done this. As long as the
+# scripts are all fast it probably doesn't matter.
+#
+# While I've written this script to be careful about spaces, spaces in
+# filenames aren't supported. Don't do that; it just creates headaches
+# for everyone.
+#
+
+# Get the list of tests.
+#
+# Note that in some shells (and depending on settings) asking for
+# *.saw when there aren't any will yield "*.saw" rather than
+# generating an error.
+TESTS=
+for SCRIPT in *.saw; do
+    BASE=${SCRIPT%.saw}
+    TESTS="$TESTS $BASE"
+done
+if [ "$TESTS" = " *.saw" ] || [ "$TESTS" = " " ]; then
+    echo "$0: Found no files matching *.saw" 1>&2
+    exit 1
+fi
+
+# Get the current directory, because we'll need to filter it out of
+# the saw output. Be sure to get the real current directory. Because
+# ksh's silly ideas about symlinks got stuffed into posix, pwd can lie
+# and then what it prints won't be the same as what saw outputs.
+#
+# Don't use the variable PWD for this; it has magic semantics in the
+# shell and, for the same reasons, may also lie.
+#
+# Bomb if the current directory includes a comma (just in case) so we
+# can use comma as a delimiter in the seddery below.
+CURDIR=$(pwd -P || pwd)
+if echo "$CURDIR" | grep -q , >/dev/null; then
+    echo "$0: Your current directory name contains a comma." 1>&2
+    echo "$0: I can't deal with that. Sorry..." 1>&2
+    exit 1
+fi
+
+# shell function for the run-tests op
+run-tests() {
+    for TEST in $TESTS; do
+        # Remove any existing test.log first as a precaution. This
+        # protects against misreading the results if the whole run
+        # gets killed before a new test.log gets produced.
+        rm -f $TEST.log
+
+        # run the test
+        # (do not fail if saw does, instead log it)
+        echo "$SAW $TEST.saw"
+        $SAW $TEST.saw > $TEST.rawlog 2>&1 || echo FAILED >> $TEST.rawlog
+
+        # Prune the timestamps from the log since they'll never match.
+        # We also need to shorten pathnames that contain the current
+        # directory, because saw feels the need to expand pathnames
+        # (see also saw-script #2082).
+        sed < $TEST.rawlog > $TEST.log '
+            /^\[[0-9][0-9]:[0-9][0-9]:[0-9][0-9]\.[0-9][0-9][0-9]\] /{
+                s/^..............//
+            }
+            s,'"$CURDIR"'/,,
+        '
```

Comment:
Ideally, we'd introduce a setting into SAW that omits timestamps so that we don't have to strip them out afterwards. Can you file an issue about this? (No need to hold up this PR on that feature, but I want to make sure the idea is recorded.)

Reply:
I filed #2090; #2082 is also relevant to the pathnames issue.